cmd/brew-test-bot.rb

```diff
@@ -1249,6 +1266,10 @@ def test_ci_upload(tap)
 
     # These variables are for Jenkins, Circle CI, and Travis CI respectively.
     upstream_number = ENV["UPSTREAM_BUILD_NUMBER"] || ENV["CIRCLE_BUILD_NUM"] || ENV["TRAVIS_BUILD_NUMBER"]
+    # If the GITHUB_TOKEN variable is set, use token+https instead of ssh for pushing.
+    # Note that we do not use 'tap.user' but instead GIT_AUTHOR_NAME. This is because
+    # we may push to a different place than the 'tap.user' repo. For example,
+    # for Homebrew-core, tap.user=Homebrew but GIT_AUTHOR_NAME=BrewTestBot.
     remote = if github_token
       "https://#{github_token}@github.com/#{ENV["GIT_AUTHOR_NAME"]}/homebrew-#{tap.repo}.git"
```

Comment:
Upstream they use the var `HOMEBREW_GIT_NAME` at this same line. Is it different from `GIT_AUTHOR_NAME `? `GIT_AUTHOR_NAME ` is not set by --git-name